reposerver/repository/repository.proto

```diff
@@ -256,6 +256,28 @@ message GitDirectoriesResponse {
     repeated string paths = 1;
 }
 
+message UpdateRevisionForPathsRequest {
+    github.com.argoproj.argo_cd.v2.pkg.apis.application.v1alpha1.Repository repo = 1;
+
+    string appLabelKey = 2;
+    string appName = 3;
+    string namespace = 4;
+    github.com.argoproj.argo_cd.v2.pkg.apis.application.v1alpha1.ApplicationSource applicationSource = 5;
+    string trackingMethod = 6;
+    map<string, github.com.argoproj.argo_cd.v2.pkg.apis.application.v1alpha1.RefTarget> refSources = 7;
+    string kubeVersion = 8;
+    repeated string apiVersions = 9;
+    bool hasMultipleSources = 10;
+
+    string syncedRevision = 11;
+    string revision = 12;
+    repeated string paths = 13;
+}
+
+message UpdateRevisionForPathsResponse {
+    bool changes = 1;
```

Comment:
I'm wondering if we should just drop this field. It's not used anywhere besides the unit test. If what we really need from the API call is the side effect (updating the cache key), then we should probably just test for the side-effect.